HIP/hip-435.md

```diff
@@ -0,0 +1,196 @@
+
+### Record Stream File Changes
+
+A new version (V6) of the record stream file format will be created using the following protobufs:
+
+```
+message RecordFileObject {
+  Transaction transaction = 1;
+  TransactionRecord record = 2;
+}
+```
+
+`Transaction` and `TransactionRecord` are already defined as protobuf messages and are used by the Record Stream algorithm. No changes will be required for these messages.
+
+`RecordFileObjects` are serialized in record stream files.
+
+```
+message RecordStreamFile {
+
+  // Version of HAPI that was used to serialise the file
+  SemanticVersion hapiProtoVersion = 1;
+
+  // Running Hash of all RecordFileObjects before writing this file.
+  HashObject startObjectRunningHash = 2; 
+
+  // List of all the record file objects from that period.
+  repeated RecordFileObject fileObjects = 3; 
+
+  // Running Hash of all RecordFileObjects before closing this file.
+  HashObject endObjectRunningHash = 4; 
+
+  // The block number associated with this period.
+  int64 blockNumber = 5; 
+
+  // List of the hashes of all the Sidecar record files created for the same period.
+  // Allows multiple sidecar files to be linked to this RecordStreamFile
+  repeated bytes SidecarFileHash = 6; 
+}
+```
+
+```
+message HashObject {
+  enum HashAlgorithm {
+    SHA-384 = 0
+  }
+  HashAlgorithm algorithm = 0;
+  int length = 1;
+  bytes hash = 2;
+}
+```
+
+`RecordStreamFile` is a newly defined protobuf message which will be used to serialise all `RecordFileObject`s that are part of the same period into record stream files. Periods are intervals of `hedera.recordStream.logPeriod` seconds used to group `RecordFileObject`s into record stream files.
+The record signature file that is created for each record stream file signs the hash of the entire corresponding stream file. The list of sidecar file hashes is included in the record stream file. This way mirror nodes or any interested party can download the record stream file and all sidecar files and verify that:
+1. repeated bytes SidecarFileHash is correct; 
+2. the signature file signed the correct hash of the entire record stream file;
+
+```
+message TransactionSidecarRecord {
+    // Consensus timestamp will be the same as the consensus timestamp of the 
+    // transaction the side car is related to. This offers a convenient 
+    // way to match record to sidecar. 
+    Timestamp consensusTimestamp = 1;
+    oneof {
+      repeated ContractStateChange = 2; // Example sidecar type
+      repeated ContractAction = 3; // Example sidecar type
+      // Other future categories
+    }
+}
+```
+
+`TransactionSidecarRecord` is a protobuf message which will be used to create sidecar records complementing `TransactionRecord` and storing additional information about a transaction execution. 
+
+```
+message SidecarFile {
+  // Version of HAPI that was used to serialise the file
+  SemanticVersion hapiProtoVersion = 1;
```

Comment:
Don't think this is needed since the parent RecordStreamFile has this info already